src/controllers/roleController.js

```diff
@@ -0,0 +1,207 @@
+        } catch (error) {
+            res.status(400).json({ statusCode: 400, data: error });
+        }
+    }
+
+    const getOneRole = async(req, res) => {
+        const id = req.params.id;
+        try {
+            const roles = await role.findOne({
+                where: { id },
+            });
+            if (roles) {
+                res.status(200).json({ message: req.t('role_fetch_message'), data: roles });
+            } else {
+                res.status(404).json({
+                    statusCode: 404,
+                    Message: req.t('role_unexist_message'),
+                });
+            }
+        } catch (error) {
+            res.status(400).json({ statusCode: 400, data: error });
+        }
+    }
+
+    const deleteOneRole = async(req, res) => {
+        const id = req.params.id;
+        const roles = await role.findOne({
+            where: { id},});
+        try {
+            if (roles) {
+                await role.destroy({
+                    where: {
+                        id
+                    },});
+                res.status(200).json({
+                    statusCode: 200,
+                    message: req.t('role_delete_message')
+                    
+                });
+            } else {
+                res.status(404).json({
+                    statusCode: 404,
+                    Message: req.t('role_unexist_message'),
+                });}
+        } catch (error) {
+            res.status(400).json({ statusCode: 400, data: error });
+        }}
+
+    const updateOneRole = async(req, res)=> {
+        const id = req.params.id;
+        const roles = await role.findOne({
+            where: { id },});
+        const { roleName, description } = req.body;
+        try {
+            if (!roles) {
+                return res.status(404).json({
+                    statusCode: 404,
+                    message: req.t('role_unexist_message'),
+                });
+            } else {    
+                await role.update(
+                    { roleName, description },
+                    {
+                        where: {id},});
+                return res
+                    .status(201)
+                    .json({ statusCode: 201, message: req.t('role_update_message'),data:role });}
+        } catch (error) {
+            console.error(error);
+            return res
+                .status(500)
+                .json({ statusCode: 400, error: "Server error" });
+        }}
+
+    const setRole = async (req, res) => {
+        const roleData = {
+            userId: req.body.userId,
+            roleId: req.body.roleId,
+        }
+        if (!userId) {
+            return res.status(404).send(`${userId} is not found`);
+          }
+        try {
+            const userRoles = await userRole.create(roleData)
+            return res.status(201).json({ message: req.t('set-role-message'), response: userRoles }) 
```

Comment:
snake case would be good instead `set-role-message` you can use `set_role_message`